documentation/modules/oauth/proc-oauth-authorization-keycloak-example.adoc

```diff
@@ -239,14 +250,24 @@ SSO_HOST=_SSO-HOSTNAME_
 SSO_HOST_PORT=$SSO_HOST:443
 STOREPASS=storepass
 
-echo "Q" | openssl s_client -showcerts -connect $SSO_HOST_PORT 2>/dev/null | awk ' /BEGIN CERTIFICATE/,/END CERTIFICATE/ { print $0 } ' > /tmp/sso.crt
+echo "Q" | openssl s_client -showcerts -connect $SSO_HOST_PORT 2>/dev/null | awk ' /BEGIN CERTIFICATE/,/END CERTIFICATE/ { print $0 } ' > /tmp/sso.pem
 ----
-
++
+Usually there is not one single certificate, but a certificate chain. You only have to provide the top-most issuer CA which is listed as last in the `/tmp/sso.pem`.
+You can extract it manually or using the following command:
++
+[source,shell,subs="+quotes"]
+----
+split -p "-----BEGIN CERTIFICATE-----" sso.pem sso-
+for f in $(ls sso-*); do mv $f $f.pem; done
+cp $(ls sso-* | sort -r | head -n 1) sso-ca.crt
+----
++
 . Create a truststore for TLS connection to the Kafka brokers.
 +
 [source,shell,subs="+quotes"]
 ----
-keytool -keystore /tmp/truststore.p12 -storetype pkcs12 -alias sso -storepass $STOREPASS -import -file /tmp/sso.crt -noprompt
+keytool -keystore /tmp/truststore.p12 -storetype pkcs12 -alias sso -storepass $STOREPASS -import -file /tmp/sso-ca.crt -noprompt
 ----
 
 . Use the Kafka bootstrap address as the hostname of the Kafka broker and the `tls` listener port (9093) to prepare a certificate for the Kafka broker.
```

Comment:
Unlike with the OAuth server, we actually know pretty well how the TLS is designed for the Kafka cluster since that is our work. The user should either use the Cluster CA public key or the CA used to sign the custom listener certificate if they use it. Such a description would be much more reliable than trying to dig the broker certificates through `openssl`.

Reply:
It is reliable to the extent that you can simply map `my-cluster-clients-ca-cert` secret by keys `ca.p12` and `ca.password` to the env vars in the yaml.

Here in the example we are inside the shell, as we use the Kafka CLI tools, and  in order to show how one would do it even from outside the cluster.
```
NS=sso
kubectl run -ti --restart=Never --image={DockerKafkaImageCurrent} kafka-cli -n $NS -- /bin/sh
```
I bet there are other places in the docs that clearly describe how to mount the secret for your pod and I prefer to simply link to those descriptions, and to emphasise in the documentation that using `openssl` to get the certificate is not a proper safe way. Rather one should get the trusted certificate from kubernetes cluster or from cluster administrator.

Reply:
But what you have here basically works in like 20% of cases only. Everyone else will be just confused as well about why the super detailed guide does not work.

Reply:
What makes you think it only works for 20% of the cases?

SSO is pretty simple - it's TLS protected web server. All we need is a local truststore with the CA.
How does the above procedure not work for someone?

Reply:
I do not care about the SSO part. I talk about the Kafka part in all my comments